\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={getminipackage.R},
            pdfauthor={benjk},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{getminipackage.R}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{benjk}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2019-05-06}


\begin{document}
\maketitle

getBiotypes function

@aliases getBiotypes

@description The purpose of the getBiotypes function is to class both
coding and noncoding transcripts into biotypes using the most recent
GENCODE annotations. This tool can also be used to define potential
lncRNAs, given an available genome transcriptome assembly (a gtf file)
or any genomic loci of interest.

@param full\_gr A GRanges object of coding and noncoding transctipts.
Unique identifications for each column must be assigned.More details can
be found in the GRanges package. @param gencode\_gr This GRanges object
contains a GENCODE reference annotation.It must have a column of
biotypes. @param intron\_gr A GRanges object containing the coordinates
of introns.For details see GRanges package. @param minoverlap Detects
minimum overlap between two IRanges objects. Details Overlap arguments
are included in the IRanges package.

@details For details of findOverlaps, type.partialOverlap,
type.50Overlap type.toPlot, queryhits, and subjecthits see
\href{https://www.bioconductor.org/packages/release/bioc/html/GenomicRanges.html}{GenomicRanges}
\href{https://www.bioconductor.org/packages/release/bioc/html/IRanges.html}{IRanges},
and \href{http://bioconductor.org/install/index.html}{BiocManager}.

@return Returns the classified transcriptome biotypes.

@source
\href{https://www.gencodegenes.org/human/release_25lift37.html}{Refrence
GRCh37 genome} for details on gtf format visit
\href{https://useast.ensembl.org/info/website/upload/gff.html}{ensemble}
@import GenomicRanges

@references

Wang, Z. Z., J. M. Cunningham and X. H. Yang (2018).'CisPi: a
transcriptomic score for disclosing cis-acting disease-associated
lincRNAs.' Bioinformatics34(17): 664-670'

@examples \#Input datasets locally data(``gencode\_gr.v19\_chr21.rda'')
data(``intron\_gr.chr21.rda'') data(``data/ILEF\_gr.chr21.rda'')

getBiotypes(ILEF\_gr,gencode\_gr)

\dontrun{getBiotypes('intron_gr')}

@note Replace the path\_file when loading data locally to the data
directory.

@import GenomicRanges IRanges @importFrom stats aggregate @export

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{getBiotypes <-}\StringTok{ }\ControlFlowTok{function}\NormalTok{(full_gr, gencode_gr, }\DataTypeTok{intron_gr =} \OtherTok{NULL}\NormalTok{, }\DataTypeTok{minoverlap =}\NormalTok{ 1L) \{}
    \CommentTok{## check input format ##########}
    \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{class}\NormalTok{(full_gr) }\OperatorTok{!=}\StringTok{ "GRanges"}\NormalTok{)}
        \KeywordTok{stop}\NormalTok{(}\StringTok{"please give full_gr as a }\CharTok{\textbackslash{}"}\StringTok{GRanges}\CharTok{\textbackslash{}"}\StringTok{ object"}\NormalTok{)}
    \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{class}\NormalTok{(gencode_gr) }\OperatorTok{!=}\StringTok{ "GRanges"}\NormalTok{)}
        \KeywordTok{stop}\NormalTok{(}\StringTok{"pealse give gencode_gr as a }\CharTok{\textbackslash{}"}\StringTok{GRanges}\CharTok{\textbackslash{}"}\StringTok{ object"}\NormalTok{)}
    \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{class}\NormalTok{(intron_gr) }\OperatorTok{!=}\StringTok{ "GRanges"} \OperatorTok{&}\StringTok{ }\OperatorTok{!}\KeywordTok{is.null}\NormalTok{(intron_gr))}
        \KeywordTok{stop}\NormalTok{(}\StringTok{"please give intron_gr as a }\CharTok{\textbackslash{}"}\StringTok{GRanges}\CharTok{\textbackslash{}"}\StringTok{ object"}\NormalTok{)}
    \CommentTok{## find transcripts overlap with any GENCODe annotated transcripts ##########}
\NormalTok{    hits =}\StringTok{ }\KeywordTok{findOverlaps}\NormalTok{(full_gr, gencode_gr, }\DataTypeTok{type =} \StringTok{"within"}\NormalTok{, }\DataTypeTok{minoverlap =}\NormalTok{ minoverlap)}
    \CommentTok{## derive the unique index from full_gr}
\NormalTok{    full =}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(full_gr)}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.fullOverlap =}\StringTok{ "de novo"}
\NormalTok{    idx =}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{mcols}\NormalTok{(full_gr[}\KeywordTok{queryHits}\NormalTok{(hits)]))}
    \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{nrow}\NormalTok{(idx) }\OperatorTok{!=}\StringTok{ }\DecValTok{0}\NormalTok{) \{}
\NormalTok{        idx}\OperatorTok{$}\NormalTok{biotype =}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{mcols}\NormalTok{(gencode_gr[}\KeywordTok{subjectHits}\NormalTok{(hits)]))[, }\DecValTok{1}\NormalTok{]}
\NormalTok{        idx_collapse =}\StringTok{ }\KeywordTok{aggregate}\NormalTok{(}\KeywordTok{as.list}\NormalTok{(idx[}\StringTok{"biotype"}\NormalTok{]), idx[}\StringTok{"Row.names"}\NormalTok{], }\DataTypeTok{FUN =} \ControlFlowTok{function}\NormalTok{(X) }\KeywordTok{paste}\NormalTok{(}\KeywordTok{unique}\NormalTok{(X),}
            \DataTypeTok{collapse =} \StringTok{", "}\NormalTok{))}
\NormalTok{        idx_full =}\StringTok{ }\KeywordTok{match}\NormalTok{(idx_collapse}\OperatorTok{$}\NormalTok{Row.names, full}\OperatorTok{$}\NormalTok{Row.names)}
\NormalTok{        full[idx_full, ]}\OperatorTok{$}\NormalTok{type.fullOverlap =}\StringTok{ }\NormalTok{idx_collapse}\OperatorTok{$}\NormalTok{biotype}
\NormalTok{    \}}
\NormalTok{    hits =}\StringTok{ }\KeywordTok{findOverlaps}\NormalTok{(full_gr, gencode_gr, }\DataTypeTok{minoverlap =}\NormalTok{ minoverlap)}
\NormalTok{    overlaps <-}\StringTok{ }\KeywordTok{pintersect}\NormalTok{(full_gr[}\KeywordTok{queryHits}\NormalTok{(hits)], gencode_gr[}\KeywordTok{subjectHits}\NormalTok{(hits)])}
\NormalTok{    percentOverlap <-}\StringTok{ }\KeywordTok{width}\NormalTok{(overlaps)}\OperatorTok{/}\KeywordTok{width}\NormalTok{(gencode_gr[}\KeywordTok{subjectHits}\NormalTok{(hits)])}

\NormalTok{    idx =}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{mcols}\NormalTok{(full_gr[}\KeywordTok{queryHits}\NormalTok{(hits)]))}
\NormalTok{    idx}\OperatorTok{$}\NormalTok{biotype =}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{mcols}\NormalTok{(gencode_gr[}\KeywordTok{subjectHits}\NormalTok{(hits)]))}
\NormalTok{    idx_collapse =}\StringTok{ }\KeywordTok{aggregate}\NormalTok{(}\KeywordTok{as.list}\NormalTok{(idx[}\StringTok{"biotype"}\NormalTok{]), idx[}\StringTok{"Row.names"}\NormalTok{], }\DataTypeTok{FUN =} \ControlFlowTok{function}\NormalTok{(X) }\KeywordTok{paste}\NormalTok{(}\KeywordTok{unique}\NormalTok{(X),}
        \DataTypeTok{collapse =} \StringTok{", "}\NormalTok{))}

\NormalTok{    full}\OperatorTok{$}\NormalTok{type.partialOverlap =}\StringTok{ "de novo"}
\NormalTok{    idx_partial =}\StringTok{ }\KeywordTok{match}\NormalTok{(idx_collapse}\OperatorTok{$}\NormalTok{Row.names, full}\OperatorTok{$}\NormalTok{Row.names)}
\NormalTok{    full[idx_partial, ]}\OperatorTok{$}\NormalTok{type.partialOverlap =}\StringTok{ }\NormalTok{idx_collapse}\OperatorTok{$}\NormalTok{biotype}

\NormalTok{    idx}\OperatorTok{$}\NormalTok{percentOverlap =}\StringTok{ }\NormalTok{percentOverlap}
\NormalTok{    idx_}\DecValTok{50}\NormalTok{ =}\StringTok{ }\KeywordTok{subset}\NormalTok{(idx, percentOverlap }\OperatorTok{>=}\StringTok{ }\FloatTok{0.5}\NormalTok{)}
\NormalTok{    idx_50collapse =}\StringTok{ }\KeywordTok{aggregate}\NormalTok{(}\KeywordTok{as.list}\NormalTok{(idx_}\DecValTok{50}\NormalTok{[}\StringTok{"biotype"}\NormalTok{]), idx_}\DecValTok{50}\NormalTok{[}\StringTok{"Row.names"}\NormalTok{], }\DataTypeTok{FUN =} \ControlFlowTok{function}\NormalTok{(X) }\KeywordTok{paste}\NormalTok{(}\KeywordTok{unique}\NormalTok{(X),}
        \DataTypeTok{collapse =} \StringTok{", "}\NormalTok{))}

\NormalTok{    full}\OperatorTok{$}\NormalTok{type}\FloatTok{.50}\NormalTok{Overlap =}\StringTok{ "de novo"}
\NormalTok{    idx_}\DecValTok{50}\NormalTok{ =}\StringTok{ }\KeywordTok{match}\NormalTok{(idx_50collapse}\OperatorTok{$}\NormalTok{Row.names, full}\OperatorTok{$}\NormalTok{Row.names)}
\NormalTok{    full[idx_}\DecValTok{50}\NormalTok{, ]}\OperatorTok{$}\NormalTok{type}\FloatTok{.50}\NormalTok{Overlap =}\StringTok{ }\NormalTok{idx_50collapse}\OperatorTok{$}\NormalTok{biotype}

    \ControlFlowTok{if}\NormalTok{ (}\OperatorTok{!}\KeywordTok{is.null}\NormalTok{(intron_gr)) \{}
        \CommentTok{## check introns ##########}
\NormalTok{        hits =}\StringTok{ }\KeywordTok{findOverlaps}\NormalTok{(full_gr, intron_gr)}
\NormalTok{        idx =}\StringTok{ }\KeywordTok{unique}\NormalTok{(}\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{mcols}\NormalTok{(full_gr[}\KeywordTok{queryHits}\NormalTok{(hits)])))}
\NormalTok{        full}\OperatorTok{$}\NormalTok{hasIntron =}\StringTok{ "no"}
\NormalTok{        idx_intron =}\StringTok{ }\KeywordTok{match}\NormalTok{(idx}\OperatorTok{$}\NormalTok{Row.names, full}\OperatorTok{$}\NormalTok{Row.names)}
        \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{length}\NormalTok{(idx_intron) }\OperatorTok{!=}\StringTok{ }\DecValTok{0}\NormalTok{)}
\NormalTok{            full[idx_intron, ]}\OperatorTok{$}\NormalTok{hasIntron =}\StringTok{ "yes"}
\NormalTok{    \} }\ControlFlowTok{else}\NormalTok{ (full}\OperatorTok{$}\DataTypeTok{hasIntron =} \OtherTok{NA}\NormalTok{)}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{ifelse}\NormalTok{(full}\OperatorTok{$}\NormalTok{hasIntron }\OperatorTok{==}\StringTok{ "yes"} \OperatorTok{&}\StringTok{ }\NormalTok{full}\OperatorTok{$}\NormalTok{type}\FloatTok{.50}\NormalTok{Overlap }\OperatorTok{==}\StringTok{ "protein_coding"}\NormalTok{, }\StringTok{"protein_coding_intron"}\NormalTok{,}
\NormalTok{        full}\OperatorTok{$}\NormalTok{type}\FloatTok{.50}\NormalTok{Overlap)}
    \CommentTok{## grouping into 11 biotypes ##########}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(full}\OperatorTok{$}\NormalTok{type.toPlot, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{grepl}\NormalTok{(}\StringTok{"protein_coding"}\NormalTok{, x) }\OperatorTok{&}\StringTok{ }\KeywordTok{grepl}\NormalTok{(}\StringTok{"antisense"}\NormalTok{,}
\NormalTok{        x), }\StringTok{"protein_coding_antisense"}\NormalTok{, x))}

\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(full}\OperatorTok{$}\NormalTok{type.toPlot, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{grepl}\NormalTok{(}\StringTok{"protein_coding,"}\NormalTok{, x), }\StringTok{"protein_coding_mixed"}\NormalTok{,}
\NormalTok{        x))}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(full}\OperatorTok{$}\NormalTok{type.toPlot, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{grepl}\NormalTok{(}\StringTok{", protein_coding"}\NormalTok{, x), }\StringTok{"protein_coding_mixed"}\NormalTok{,}
\NormalTok{        x))}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(full}\OperatorTok{$}\NormalTok{type.toPlot, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{grepl}\NormalTok{(}\StringTok{"lincRNA"}\NormalTok{, x), }\StringTok{"lincRNA"}\NormalTok{,}
\NormalTok{        x))}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(full}\OperatorTok{$}\NormalTok{type.toPlot, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{grepl}\NormalTok{(}\StringTok{"antisense,"}\NormalTok{, x), }\StringTok{"antisense"}\NormalTok{,}
\NormalTok{        x))}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(full}\OperatorTok{$}\NormalTok{type.toPlot, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(}\KeywordTok{grepl}\NormalTok{(}\StringTok{", antisense"}\NormalTok{, x), }\StringTok{"antisense"}\NormalTok{,}
\NormalTok{        x))}

\NormalTok{    label =}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"protein_coding"}\NormalTok{, }\StringTok{"protein_coding_mixed"}\NormalTok{, }\StringTok{"lincRNA"}\NormalTok{, }\StringTok{"antisense"}\NormalTok{, }\StringTok{"pseudogene, processed_pseudogene"}\NormalTok{,}
        \StringTok{"pseudogene, unprocessed_pseudogene"}\NormalTok{, }\StringTok{"de novo"}\NormalTok{, }\StringTok{"protein_coding_antisense"}\NormalTok{, }\StringTok{"protein_coding_intron"}\NormalTok{,}
        \StringTok{"miRNA"}\NormalTok{)}
\NormalTok{    full}\OperatorTok{$}\NormalTok{type.toPlot =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(full}\OperatorTok{$}\NormalTok{type.toPlot, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{ifelse}\NormalTok{(}\OperatorTok{!}\NormalTok{x }\OperatorTok{%in%}\StringTok{ }\NormalTok{label, }\StringTok{"other_noncoding"}\NormalTok{,}
\NormalTok{        x))}
    \KeywordTok{return}\NormalTok{(full)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

getReadthrough function

@description A getReadthrough functions that will The getReadthrough
function is used to find long transcripts that covers more than two
coding regions of a genome. The variables are as follows.

@param gr A GRanges object that shows the start and end loci on genome.
@param cod\_gr A GRanges object that contains coding regions. For
details please visit /R/data.R.

@details For details of findOverlaps, type.partialOverlap,
type.50Overlap type.toPlot, queryhits, and subjecthits see,
\href{https://www.bioconductor.org/packages/release/bioc/html/GenomicRanges.html}{GenomicRanges},
\href{https://www.bioconductor.org/packages/release/bioc/html/IRanges.html}{IRanges},
and \href{http://bioconductor.org/install/index.html}{BiocManager}.

@return Returns the classified transcriptome biotypes.

@source
\href{https://www.gencodegenes.org/human/release_25lift37.html}{Refrence
GRCh37 genome} for details on gtf format visit
\href{https://useast.ensembl.org/info/website/upload/gff.html}{ensemble}

@import GenomicRanges @importFrom stats aggregate

@references Wang, Z. Z., J. M. Cunningham and X. H. Yang (2018).'CisPi:
a transcriptomic score for disclosing cis-acting disease-associated
lincRNAs.' Bioinformatics34(17): 664-670'

@examples data(`gencode\_gr.v19\_chr21.rda') data(`ILEF\_gr.chr21.rda')
data(`cod\_gr.chr21.rda') getReadthrough(ILEF\_gr,cod\_gr)

\dontrun{getReadthrough(cod_gr)}

@note Replace the path\_file when loading data locally to the data
directory.

@import GenomicRanges IRanges @importFrom stats aggregate @export

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#gencode_gr <- data("gencode_gr.v19_chr21")}
\CommentTok{#cod_gr <- subset(gencode_gr, biotype = "protein_coding")}
\CommentTok{#cod_gr <- data("cod_gr.chr21")}
\CommentTok{#}
\NormalTok{getReadthrough <-}\StringTok{ }\ControlFlowTok{function}\NormalTok{(gr, cod_gr) \{}
\NormalTok{    full_table =}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(gr)}
\NormalTok{    overlapcount =}\StringTok{ }\KeywordTok{countOverlaps}\NormalTok{(gr, cod_gr)}
\NormalTok{    completeoverlap =}\StringTok{ }\KeywordTok{unique}\NormalTok{(}\KeywordTok{subjectHits}\NormalTok{(}\KeywordTok{findOverlaps}\NormalTok{(cod_gr, }\KeywordTok{GRanges}\NormalTok{(full_table}\OperatorTok{$}\NormalTok{ID), }\DataTypeTok{type =} \StringTok{"within"}\NormalTok{)))}
    \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{length}\NormalTok{(completeoverlap) }\OperatorTok{==}\StringTok{ }\DecValTok{0}\NormalTok{) \{}
\NormalTok{        full_table}\OperatorTok{$}\NormalTok{readthrough =}\StringTok{ }\KeywordTok{ifelse}\NormalTok{(overlapcount }\OperatorTok{>}\StringTok{ }\DecValTok{2}\NormalTok{, }\DecValTok{1}\NormalTok{, }\DecValTok{0}\NormalTok{)}
\NormalTok{    \} }\ControlFlowTok{else}\NormalTok{ \{}
\NormalTok{        full_table}\OperatorTok{$}\NormalTok{readthrough =}\StringTok{ }\KeywordTok{ifelse}\NormalTok{(overlapcount }\OperatorTok{>}\StringTok{ }\DecValTok{2} \OperatorTok{&}\StringTok{ }\KeywordTok{row.names}\NormalTok{(completeoverlap) }\OperatorTok{%in%}\StringTok{ }\NormalTok{completeoverlap,}
                                        \DecValTok{1}\NormalTok{, }\DecValTok{0}\NormalTok{)}
\NormalTok{    \}}
\NormalTok{    gr =}\StringTok{ }\KeywordTok{GRanges}\NormalTok{(}\KeywordTok{subset}\NormalTok{(full_table, readthrough }\OperatorTok{==}\StringTok{ }\DecValTok{1}\NormalTok{))}
\NormalTok{    idx =}\StringTok{ }\KeywordTok{subset}\NormalTok{(full_table, readthrough }\OperatorTok{==}\StringTok{ }\DecValTok{1}\NormalTok{)}\OperatorTok{$}\NormalTok{ID}
\NormalTok{    overlaps =}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(}\KeywordTok{findOverlaps}\NormalTok{(gr, cod_gr))}
\NormalTok{    splitoverlaps =}\StringTok{ }\KeywordTok{split}\NormalTok{(overlaps, }\DataTypeTok{f =}\NormalTok{ overlaps}\OperatorTok{$}\NormalTok{queryHits)}
    \KeywordTok{table}\NormalTok{(}\KeywordTok{sapply}\NormalTok{(splitoverlaps, nrow) }\OperatorTok{>}\StringTok{ }\DecValTok{1}\NormalTok{)}
\NormalTok{    cod_grL =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(splitoverlaps, }\ControlFlowTok{function}\NormalTok{(x) cod_gr[x}\OperatorTok{$}\NormalTok{subjectHits])}
\NormalTok{    overlapL =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(cod_grL, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{findOverlaps}\NormalTok{(x))}
\NormalTok{    notoverlap =}\StringTok{ }\KeywordTok{sapply}\NormalTok{(overlapL, }\ControlFlowTok{function}\NormalTok{(x) }\KeywordTok{identical}\NormalTok{(}\KeywordTok{queryHits}\NormalTok{(x), }\KeywordTok{subjectHits}\NormalTok{(x)))}
\NormalTok{    full_table}\OperatorTok{$}\NormalTok{readthrough =}\StringTok{ }\KeywordTok{ifelse}\NormalTok{(full_table}\OperatorTok{$}\NormalTok{readthrough }\OperatorTok{==}\StringTok{ }\DecValTok{1} \OperatorTok{&}\StringTok{ }\OperatorTok{!}\NormalTok{notoverlap, }\DecValTok{1}\NormalTok{, }\DecValTok{0}\NormalTok{)}
    \KeywordTok{return}\NormalTok{(full_table)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}


\end{document}
